### PR TITLE
Fix valid-example for $ref useage

### LINKED ISSUE #109: Spectral OAS2 validation throws a non-existent error without path value
Running spectral OAS2 rules and functions sets on a OAS2 `.yaml` file on all tests (2 different `yaml files`) results in a cryptic error without a `"path"` parameter:
```
{
    "name": "oas2-schema",
    "summary": "Validate structure of OpenAPIv2 specification.",
    "message": "should NOT have additional properties: 0",
    "path": [],
    "severity": 50,
    "severityLabel": "error"
}
```
Those files don't throw any errors in any other validation tool (swagger editor, apiary, etc.). I am happy to provide them if needed.

### **I'm submitting a...**
  - [X] bug report
  - [ ] feature request

### What is the current behavior?

Running:
```
spectral.addFunctions(oas2Functions());
spectral.addRules(oas2Rules());
const results = spectral.run(myOAS);
```
on a OAS2 `.yaml` file

### What is the expected behavior?
Do not throw this error or be more specific on it's origin and place in the file.

### What is the motivation / use case for changing the behavior?
User experience and compliance with OAS2 specification

### Please tell us about your environment:

  - Version: @stoplight/spectral@1.2.0
  - Framework: [ ]
  - Language: [all]

### Other information

It also throws 
```
oasOpParams expects a resolved object, but none was provided. Results may not be correct.
oasPathParam expects a resolved object, but none was provided. Results may not be correct.
```
which does not say anything of value without more explanation.

### PR DESCRIPTION
**Pull request by philsturgeon**
### Please check if the PR fulfills these requirements
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?

Fixing valid-examples so that examples with $ref in it, or schemas with $ref in it, will work properly.

### What is the current behavior?

Running valid-example on anything with $ref in it will cause lots of warnings to be thrown to the console, and invalid examples could sneak through. 

### Other information

Fixes #109

### CODE REVIEW

File: src/runner.ts [hunk 1]
```diff
@@ -74,6 +74,7 @@ const runRule = (
       }
     } catch (e) {
       console.warn(`Encountered error when running rule '${rule.name}' on node at path '${node.path}':\n${e}`);
+      console.error(e);
```
Comment (P0lip) [inline src/runner.ts]:
```suggestion
```

**Comment by philsturgeon** [comment]
Accidentally stole a commit from @P0lip here, but he crushed it once again!

File: src/functions/schema-path.ts [hunk 2]
```diff
@@ -13,24 +13,29 @@ import { schema } from './schema';
...
-  // Hunt in the original targetVal for the schema we are going to apply
-  let schemaObj;
+  // The subsection of the targetValue which contains the schema for us to validate the good bit against
+  let schemaObject;
   try {
-    schemaObj = jp.value(targetVal, opts.schemaPath);
+    schemaObject = jp.value(object, opts.schemaPath);
   } catch (error) {
-    console.error(error);
+    console.error('issue is here', error);
```
Comment (P0lip) [inline src/functions/schema-path.ts]:
```suggestion
    console.error(error);
```
Shall we revert to the previous message?

**Comment by philsturgeon** [comment]
Tidied up the debug stuff I had left in there. Can you do one more quick check?

**Pull request merged**